tools/sklearn/main_macros.xml

```diff
@@ -104,11 +101,80 @@ def get_X_y(params, file1, file2):
   return X, y
   </token>
 
+  <token name="@GET_SEARCH_PARAMS_FUNCTION@">
+def get_search_params(params_builder, in_pipeline=False):
+
+  search_params = {}
+
+  if not in_pipeline:
+    for ep in params_builder['estimator_param_set']:
+      if ep['estimator_param'].strip() == '':
+        continue
+      lst = ep['estimator_param'].split(":")
+      assert (len(lst) == 2), "Error, make sure there is one and only one colon in search parameter input."
+      literal = lst[1].strip().replace("__sq__","'")\
+          .replace('__dq__','"').replace("__oc__", "{").replace("__cc__", "}")\
+          .replace("__ob__", "[").replace("__cb__", "]")
+      try:
+        ev = ast.literal_eval(literal)
+      except ValueError:
+        valid = ['np.', 'numpy.', 'scipy.']
+        for v in valid:
```

Comment:
I read the docs on startswith, apparently you can just write `if literal.startswith(valid)`, if you convert `valid` into a tuple.